### PR TITLE
weidu: init at 247

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

WeiDU is the main modding tool for Infinity Engine based games.

Also includes elkhound.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128722` at 3a971a65 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>elkhound</li>
    <li>weidu</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/elkhound/default.nix:29:3:
```
   |
29 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/games/weidu/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    substitute sample.Configuration Configuration \
+      --replace /usr/bin ${lib.makeBinPath [ ocaml' ]} \
+      --replace elkhound ${elkhound}/bin/elkhound
+
+    mkdir -p obj/{.depend,x86_LINUX}
+  '';
+
+  nativeBuildInputs = [ elkhound ocaml' perl which ];
+
+  buildFlags = "weidu weinstall tolower";
```
Comment (SuperSandro2000) [inline pkgs/tools/games/weidu/default.nix:43]:
```suggestion
  buildFlags = [ "weidu" "weinstall" "tolower" ];
```

File: pkgs/development/tools/elkhound/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      install -Dm444 -t $out/lib $d/*.a
+      install -Dm444 -t $out/include/$d $src/src/$d/*.h
+    done
+    install -Dm444 -t $out/share/doc/${pname} $src/src/elkhound/*.txt
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A parser generator which emits GLR parsers, either in OCaml or C++.";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/elkhound/default.nix:43]:
```suggestion
    description = "A parser generator which emits GLR parsers, either in OCaml or C++";
```

**Review by SuperSandro2000 — COMMENTED** [review]
both meta sections are missing a homepage

**Pull request merged**